include/range/v3/view/linear_distribute.hpp

```diff
@@ -0,0 +1,191 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Eric Niebler 2013-2014
+//  Copyright Gonzalo Brito Gadeschi 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#define RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#include <cmath>
+#include <limits>
+#include <type_traits>
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/view_facade.hpp>
+#include <range/v3/utility/static_const.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+
+        /// \cond
+        namespace detail
+        {
+            template <typename T>
+            struct linear_distribute_integral_view
+              : view_facade<linear_distribute_integral_view<T>, finite>
+            {
+                static_assert(std::is_integral<T>(), "");
+            private:
+                friend range_access;
+                T from_, to_, spacing_;
+                constexpr T read() const
+                {
+                    return from_;
+                }
+                constexpr bool equal(default_sentinel) const
+                {
+                  return spacing_ == T{0}? from_ != to_ : from_ == to_ + spacing_;
+                }
+                constexpr bool equal(linear_distribute_integral_view const &that) const
+                {
+                    return that.from_ == from_ && that.to_ == to_ && that.spacing_ == spacing_;
+                }
+                RANGES_CXX14_CONSTEXPR void next()
+                {
+                    if (spacing_ == T{0})
+                    {
+                      RANGES_EXPECT(to_ == from_);
+                      to_ += 1;
+                    }
+                    from_ += spacing_;
+                }
+                RANGES_CXX14_CONSTEXPR void prev()
+                {
+                    from_ -= spacing_;
+                }
+                RANGES_CXX14_CONSTEXPR void advance(std::ptrdiff_t n)
+                {
+                    
+                  RANGES_EXPECT(spacing_ == T{0} || std::numeric_limits<T>::max() / spacing_ >= n);
+                    RANGES_EXPECT(to_ - from_ >= n * spacing_);
+                    from_ += n * spacing_;
+                }
+                constexpr std::ptrdiff_t distance_to(linear_distribute_integral_view const &that) const
+                {
+                    return (that.from_ - from_) / spacing_;
+                }
+            public:
+                linear_distribute_integral_view() = default;
+                linear_distribute_integral_view(T from, T to, std::ptrdiff_t n)
+                  : from_(from), to_(to) 
+                {
+                    RANGES_EXPECT(n > 0);
```

Comment:
This is redundant with the checks in the `if` / `else`, and `to >= from_` is partially redundant. Why not replace the whole body with:
```c++
if (to_ == from_)
{
    RANGES_EXPECT(n == 1);
    spacing_ = 0;
}
else
{
    RANGES_EXPECT(n > 1);
    RANGES_EXPECT(from < to_);
    spacing = (to_ - from_) / static_cast<T>(n - 1);
}
```